### PR TITLE
Add more logger data to GitHub clients

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @cjwagner @fej

### CODE REVIEW

File: maintenance/migratestatus/migratestatus.go [hunk 1]
```diff
@@ -123,7 +124,7 @@ func main() {
 		}
 	}
 
-	githubClient, err := o.github.GitHubClient(secretAgent, o.dryRun)
+	githubClient, err := o.github.GitHubClient(secretAgent, o.dryRun, componentField)
```
Comment (cjwagner) [inline maintenance/migratestatus/migratestatus.go]: We don't need to include fields that are already included via the DefaultFieldsFormatter.

File: prow/flagutil/github.go [hunk 2]
```diff
@@ -117,7 +117,7 @@ func (o *GitHubOptions) GitClient(secretAgent *secret.Agent, dryRun bool) (clien
 	}(client)
 
 	// Get the bot's name in order to set credentials for the Git client.
-	githubClient, err := o.GitHubClient(secretAgent, dryRun)
+	githubClient, err := o.GitHubClient(secretAgent, dryRun, logrus.Fields{"component": "flagutil"})
```
Comment (cjwagner) [inline prow/flagutil/github.go]: `flagutil` is not a component. We don't want to override "component: tide" just because tide uses the `flagutil` package.

Comment (stevekuznetsov) [reply prow/flagutil/github.go]: This is an ephemeral client used for one call in this helper :|

File: maintenance/migratestatus/migratestatus.go (hunk 1 above)
Comment (stevekuznetsov) [reply maintenance/migratestatus/migratestatus.go]: D'oh, I thought it had neither :|

**Comment by stevekuznetsov** [comment]
@cjwagner @fejta made this target `tide` specifically and opt-in

File: prow/flagutil/github.go (hunk 2 above)
Comment (cjwagner) [reply prow/flagutil/github.go]: Oh I didn't see that 🤦‍♂️

**Comment by fejta** [comment]
ref https://github.com/kubernetes/test-infra/issues/11430

**Pull request merged**